src/main/java/com/cleanroommc/groovyscript/compat/mods/bloodmagic/AlchemyArray.java

```diff
@@ -0,0 +1,182 @@
+package com.cleanroommc.groovyscript.compat.mods.bloodmagic;
+
+import WayofTime.bloodmagic.api.impl.BloodMagicAPI;
+import WayofTime.bloodmagic.api.impl.recipe.RecipeAlchemyArray;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.core.mixin.bloodmagic.BloodMagicRecipeRegistrarAccessor;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.ingredient.IngredientHelper;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.ResourceLocation;
+import org.jetbrains.annotations.Nullable;
+
+public class AlchemyArray extends VirtualizedRegistry<RecipeAlchemyArray> {
+    public AlchemyArray() {
+        super("AlchemyArray", "alchemyarray", "alchemyArray", "alchemy_array");
```

Comment:
You can remove those name if your branch is up to date. Those are now auto generated based on this classes name